ospfd/ospf_routemap.c

```diff
@@ -398,17 +398,19 @@ route_set_metric(void *rule, const struct prefix *prefix, void *object)
 	if (!metric->used)
 		return RMAP_OKAY;
 
-	ei->route_map_set.metric = DEFAULT_DEFAULT_METRIC;
+	ROUTEMAP_METRIC(ei) = ei->metric;
 
 	if (metric->type == metric_increment)
-		ei->route_map_set.metric += metric->metric;
+		ROUTEMAP_METRIC(ei) += metric->metric;
 	else if (metric->type == metric_decrement)
-		ei->route_map_set.metric -= metric->metric;
+		ROUTEMAP_METRIC(ei) -= metric->metric;
 	else if (metric->type == metric_absolute)
-		ei->route_map_set.metric = metric->metric;
+		ROUTEMAP_METRIC(ei) = metric->metric;
 
-	if (ei->route_map_set.metric > OSPF_LS_INFINITY)
-		ei->route_map_set.metric = OSPF_LS_INFINITY;
+	if ((uint32_t)ROUTEMAP_METRIC(ei) < ei->min_metric)
+		ROUTEMAP_METRIC(ei) = ei->min_metric;
+	if ((uint32_t)ROUTEMAP_METRIC(ei) > ei->max_metric)
```

Comment:
Can it be higher than OSPF_LS_INFINITY?

Reply:
currently no, but now that you say it, I want to allow higher values and trim excess here since the added route map rule might be used with other protocol in the future.

Reply:
I double checked. The code is already doing the right thing. I.e,  in the context of OSPF the limit is always trimmed to `OSPF_LS_INFINITY`.  The value in the route map is still stored as-is.